es6/tests/base.js

```diff
@@ -924,3 +926,93 @@ describe("Serialization", function() {
 		JSON.stringify(doc);
 	});
 });
+
+describe("Docxtemplater v4 tests", function() {
+	it("should work when modules are attached with new constructor", function() {
+		let isModuleCalled = false;
+
+		const module = {
+			optionsTransformer(options) {
+				isModuleCalled = true;
+				return options;
+			},
+		};
+
+		createDocV4("tag-example.docx", { modules: [module] });
+		expect(isModuleCalled).to.equal(true);
+	});
+
+	it("should throw an error when modules passed is not an array", function() {
+		expect(createDocV4.bind(this, "tag-example.docx", { modules: {} })).to.throw(
+			"The modules argument of docxtemplater's constructor must be an array"
+		);
+	});
+
+	it("should throw an error when an invalid zip is passed", function() {
+		const zip = getZip("tag-example.docx");
+		zip.files = null;
+
+		expect(() => new Docxtemplater(zip)).to.throw(
+			"The first argument of docxtemplater's constructor must be a valid zip file (jszip v2 or pizzip v3)"
+		);
+
+		expect(() => new Docxtemplater("content")).to.throw(
+			"The first argument of docxtemplater's constructor must be a valid zip file (jszip v2 or pizzip v3)"
+		);
+
+		expect(() => new Docxtemplater(Buffer.from("content"))).to.throw(
+			"The first argument of docxtemplater's constructor must be a valid zip file (jszip v2 or pizzip v3)"
+		);
+	});
+
+	it("should work when the delimiters are passed with new constructor", function() {
+		const options = {
+			delimiters: {
+				start: "<",
+				end: ">",
+			},
+		};
+		const doc = createDocV4("delimiter-gt.docx", options);
+		doc.setData({
+			user: "John",
+		});
+		doc.render();
+		const fullText = doc.getFullText();
+		expect(fullText).to.be.equal("Hello John");
+	});
+
+	it("should work with options object when both modules and delimiters are passed with new constructor", function() {
```

Comment:
Can you apply the following patch for this test case : 

```
diff --git a/es6/tests/base.js b/es6/tests/base.js
index 56d0325..2b86df7 100644
--- a/es6/tests/base.js
+++ b/es6/tests/base.js
@@ -943,7 +943,9 @@ describe("Docxtemplater v4 tests", function() {
 	});
 
 	it("should throw an error when modules passed is not an array", function() {
-		expect(createDocV4.bind(this, "tag-example.docx", { modules: {} })).to.throw(
+		expect(
+			createDocV4.bind(this, "tag-example.docx", { modules: {} })
+		).to.throw(
 			"The modules argument of docxtemplater's constructor must be an array"
 		);
 	});
@@ -982,27 +984,39 @@ describe("Docxtemplater v4 tests", function() {
 	});
 
 	it("should work with options object when both modules and delimiters are passed with new constructor", function() {
-		let isModuleCalled = false;
+		let isModuleCalled = false,
+			optionsPassedToModule;
 		const options = {
 			delimiters: {
-				start: "<",
-				end: ">",
+				start: "%",
+				end: "%",
 			},
-			modules: [{
-				optionsTransformer(options) {
-					isModuleCalled = true;
-					return options;
+			modules: [
+				{
+					optionsTransformer(options) {
+						optionsPassedToModule = options;
+						isModuleCalled = true;
+						return options;
+					},
 				},
-			}],
+			],
 		};
-		const doc = createDocV4("delimiter-gt.docx", options);
+		const doc = createDocV4("delimiter-pct.docx", options);
 		doc.setData({
 			user: "John",
+			company: "Acme",
 		});
-		doc.render();
-		const fullText = doc.getFullText();
-		expect(fullText).to.be.equal("Hello John");
+
 		expect(isModuleCalled).to.be.equal(true);
+		expect(optionsPassedToModule.delimiters.start).to.be.equal("%");
+		expect(optionsPassedToModule.delimiters.end).to.be.equal("%");
+		// Verify that default options are passed to the modules
+		expect(optionsPassedToModule.linebreaks).to.be.equal(false);
+
+		doc.render();
+
+		const fullText = doc.getFullText();
+		expect(fullText).to.be.equal("Hello John from Acme");
 	});
 
 	it("should render correctly using the new constructor", () => {
```

(You can do so by running `git patch <the-file`)

Reply:
This is in order to test that the options are correctly passed to the module, because they should have access to it.

Reply:
Changing it. 👍
I did it with `git apply` because I couldn't find the `git patch` command.

Reply:
Done

Reply:
Oh yes, I meant with git apply, sorry